CMakeLists.txt

```diff
@@ -16,6 +16,7 @@ include(helpers)
 # Build Options
 option(UMF_DEVELOPER_MODE "enable developer checks, treats warnings as errors" OFF)
 option(UMF_BUILD_SHARED_LIBRARY "Build UMF as shared library" OFF)
+option(UMF_ENABLE_POOL_TRACKING "Build UMF with pool tracking" OFF)
```

Comment:
We should also move the APIs (based on tracking capabilities) mentioned below to `memory_pool.h` and add corresponding tests to make sure tracking works as expected. Also, I think we need to turn on this functionality by default.
```
///
/// \brief Frees the memory space pointed by ptr if it belongs to UMF pool, does nothing otherwise.
/// \param ptr pointer to the allocated memory
/// \return UMF_RESULT_SUCCESS on success or appropriate error code on failure.
///         Whether any status other than UMF_RESULT_SUCCESS can be returned
///         depends on the memory provider used by the pool.
///
enum umf_result_t umfFree(void *ptr);

///
/// \brief Retrieve memory pool associated with a given ptr. Only memory allocated
///        with the usage of a memory provider is being tracked.
/// \param ptr pointer to memory belonging to a memory pool
/// \return Handle to a memory pool that contains ptr or NULL if pointer does not belong to any UMF pool.
umf_memory_pool_handle_t umfPoolByPtr(const void *ptr);
```

Reply:
The API mentioned above already are in `memory_pool.h`.
Some tests already are in `test/memoryPool.hpp`.

Reply:
Turned on by default.